app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -167,26 +172,12 @@ public boolean onPreferenceClick(Preference preference) {
                 GoogleAnalyticsUtils.reportPrefItemClick(
                         GoogleAnalyticsFields.CATEGORY_DEV_PREFS,
                         GoogleAnalyticsFields.LABEL_CUSTOM_RESTORE);
-                startFileBrowser();
+                startFileBrowser(DeveloperPreferences.this, REQUEST_SYNC_FILE, "cannot.restore.xml");
```

Comment:
Nevermind ignore this comment; I didn't realize that this triggers the settings activities to finish and do the restore immediately